bundle/jsky.app.ot/src/main/java/jsky/app/ot/gemini/editor/targetComponent/EdCompTargetList.java

```diff
@@ -578,25 +587,35 @@ private void showTargetTag() {
         }
 
         @Override public void actionPerformed(final ActionEvent actionEvent) {
-            final TargetEnvironment env    = obsComp.getTargetEnvironment();
+            appendNewGroup(obsComp, positionTable);
+        }
+    }
 
-            // Ensure we are working with a guide env with a primary group.
-            final GuideEnvironment ge     = env.getGuideEnvironment();
-            final GuideGroup primaryGroup = ge.getPrimary();
+    /**
+     * Append a guide group to the collection of guide groups and select it as the primary.
+     */
+    private static Option<IndexedGuideGroup> appendNewGroup(final TargetObsComp obsComp, final TelescopePosTableWidget positionTable) {
+        final TargetEnvironment env    = obsComp.getTargetEnvironment();
 
-            final ImList<GuideGroup> oldGroups = ge.getOptions();
-            final int newGroupIdx              = oldGroups.size();
-            final GuideGroup newGroup          = GuideGroup.create("Manual Group");
-            final IndexedGuideGroup igg        = new IndexedGuideGroup(newGroupIdx, newGroup);
-            final ImList<GuideGroup> newGroups = oldGroups.append(newGroup);
+        // Ensure we are working with a guide env with a primary group.
+        final GuideEnvironment ge     = env.getGuideEnvironment();
+        final GuideGroup primaryGroup = ge.getPrimary();
 
-            // OT-34: make new group primary and select it
-            if (!positionTable.confirmGroupChange(primaryGroup, newGroup)) return;
-            obsComp.setTargetEnvironment(env.setGuideEnvironment(ge.setOptions(newGroups)
-                    .setPrimaryIndex(newGroupIdx)));
+        final ImList<GuideGroup> oldGroups = ge.getOptions();
+        final int newGroupIdx              = oldGroups.size();
+        final GuideGroup newGroup          = GuideGroup.create("Manual Group");
+        final IndexedGuideGroup igg        = new IndexedGuideGroup(newGroupIdx, newGroup);
+        final ImList<GuideGroup> newGroups = oldGroups.append(newGroup);
 
-            positionTable.selectGroup(igg);
+        // OT-34: make new group primary and select it
+        if (!positionTable.confirmGroupChange(primaryGroup, newGroup)) {
```

Comment:
Sure, I'll change it. The `if` statement was already implemented before I came along, but no reason why it shouldn't be cleaned up.